### ISSUE TITLE
Copy of tc-java#3893

### ISSUE THREAD

**Issue by DanielHabenicht** [testcontainers/testcontainers-dotnet#626]
**Describe the bug**
The mysql container does not allow the username `root` for the `MYSQL_USER`, which makes the current TestContainerConfiguration not useful as the container will fail to start. 
https://github.com/testcontainers/testcontainers-dotnet/blob/08e7fa5a6e1d08c02bf249db79ac57e097276ba6/src/Testcontainers/Configurations/Modules/Databases/MySqlTestcontainerConfiguration.cs#L40-L44

**To Reproduce**
```csharp
_container = new TestcontainersBuilder<MySqlTestcontainer>()
    .WithDatabase(new MySqlTestcontainerConfiguration { Database = "datadonation", Username = "root", Password = "example" })
    .WithImage("mysql:8.0.29")
    .WithCleanUp(true)
    .Build();
```

**Expected behavior**
Should incoporate the possibility to use `root` user (e.g. for migrations)

**Additional context**
Basically a copy of https://github.com/testcontainers/testcontainers-java/issues/3893 for dotnet.

**Comment by HofmeisterAn** [comment]
You can probably use the `MariaDbTestcontainerConfiguration`. Nevertheless, it would make sense to support it for the MySQL too. Probably it is better to inherit one configuration and adjust it to the Java behavior? Would you like to create a pull request?

**Comment by DanielHabenicht** [comment]
I don't think it would make sense to have the same configuration for two different things. 
This is also not about allowing another variable to be passed in, but about abstracting the more complicated container settings to one that works as expected from the current interface. 

Basically: 
```csharp
new MySqlTestcontainerConfiguration { 
    Username = "standarduser", 
    Password = "standardpassword"
}
```
should set: `MYSQL_USER="standarduser"` and `MYSQL_PASSWORD="standardpassword"`

```csharp
new MySqlTestcontainerConfiguration { 
    Username = "root", 
    Password = "rootpassword"
}
```
should set: `MYSQL_ROOT_PASSWORD="rootpassword"`

While still giving the right connectionString.

**Comment by DanielHabenicht** [comment]
I made a draft here #629

**Comment by HofmeisterAn** [comment]
> I don't think it would make sense to have the same configuration for two different things.

What I meant by that is, MySQL and MariaDB shares the same configuration. Right now, the configuration is copied and pasted. I think we can just inherit from one or the other and get rid of the copy and paste code.
